clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2511,6 +2620,9 @@ void Sema::CheckSYCLKernelCall(FunctionDecl *KernelFunc, SourceRange CallLoc,
     return;
 
   KernelObjVisitor Visitor{*this};
+  SYCLKernelNameTypeVisitor KernelTypeVisitor(*this, Args[0]->getExprLoc());
+  if (LangOpts.SYCLIsDevice)
```

Comment:
Should we call the whole `CheckSYCLKernelCall` function only in device mode?

Reply:
I added this check to differentiate between SYCL device kernel and SYCL host kernel based on this [PR](https://github.com/intel/llvm/pull/2186).  I am not sure if all the diagnostics emitted in `checkSYCLKernelCall()` would apply only to SYCL device kernel. If yes, I can add `if (LangOpts.SYCLIsDevice)` before calling `checkSYCLKernelCall()`

Reply:
> I am not sure if all the diagnostics emitted in checkSYCLKernelCall() would apply only to SYCL device kernel.

I think our target in `checkSYCLKernelCall()` was to diagnose problems of device kernel. However if we will need some diagnostics for host "kernel" we might use this function as well.
@premanandrao , what do you think? Do you expect any requests to diagnose misuses of function with "sycl_kernel" attribute in host mode?

Reply:
> what do you think? Do you expect any requests to diagnose misuses of function with "sycl_kernel" attribute in host mode?

I think it would be nice to be capable of diagnosing in host mode; although I am not aware of any specific requests yet.